### ISSUE TITLE
macro error: missing parameter type for expanded function

### ISSUE THREAD

**Issue by fommil** [47deg/freestyle#369]
the code in this branch/file https://github.com/fommil/drone-dynamic-agents/blob/freestyle-bug-for/src/main/scala/logic.scala

fails with

```
[error] <macro>:17: missing parameter type for expanded function ((x$1: <error>) => x$1: @scala.unchecked)
[error]     }).map((_: @scala.unchecked()) match {
[error]             ^
```

It is expanded into the code at the bottom of this report.

This could well be a `scala.meta` bug because previously the entire `trait` was not being sent to `scala.meta` as input

The shortened version is that this

```scala
  def update(old: WorldView): FS[WorldView] =
    for {
      snap    <- initial
      changed = symdiff(old.alive.keySet, snap.alive.keySet)
    } yield old
```

also fails. My guess is there is a problem with single generator `for` comprehensions.

```scala
{
  trait DynAgents[FF$1[_]] extends _root_.freestyle.internal.EffectLike[FF$1] {
    val d: Drone[FF$1]
    val m: Machines[FF$1]
    def initial: FS[WorldView] = (d.getBacklog |@| d.getAgents |@| m.getManaged |@| m.getAlive |@| m.getTime).map({
      case (db, da, mm, ma, mt) =>
        WorldView(db, da, mm, ma, Map.empty, mt)
    })
    def update(old: WorldView): FS[WorldView] = initial.map(snap => {
      val changed = symdiff(old.alive.keySet, snap.alive.keySet)
      val pending = (old.pending -- changed).filterNot({
        case (_, started) =>
          timediff(started, snap.time) >= 10.minutes
      })
      val update = snap.copy(pending = pending)
      (snap, changed, pending, update)
    }).map((_: @scala.unchecked()) match {
      case (snap, changed, pending, update) => update
    })
    private def symdiff[T](a: Set[T], b: Set[T]): Set[T] = a.union(b) -- a.intersect(b)
    def act(world: WorldView): FS[WorldView] = world match {
      case NeedsAgent(node) =>
        m.start(node).map(_ => {
          val update = world.copy(pending = Map(node -> world.time))
          (_, update)
        }).map((_: @scala.unchecked()) match {
          case (_, update) => update
        })
      case Stale(nodes) =>
        nodes.traverse(m.stop).map(stopped => {
          val updates = stopped.map(_ -> world.time).toList.toMap
          val update = world.copy(pending = world.pending ++ updates)
          (stopped, updates, update)
        }).map((_: @scala.unchecked()) match {
          case (stopped, updates, update) => update
        })
      case _ =>
        FreeS.pure(world)
    }
    private def timediff(from: ZonedDateTime, to: ZonedDateTime): FiniteDuration = ChronoUnit.MINUTES.between(from, to).minutes
    private object NeedsAgent {
      def unapply(world: WorldView): Option[Node] = world match {
        case WorldView(backlog, 0, managed, alive, pending, _) if backlog > 0 && alive.isEmpty && pending.isEmpty =>
          Option(managed.head)
        case _ =>
          None
      }
    }
    private object Stale {
      def unapply(world: WorldView): Option[NonEmptyList[Node]] = world match {
        case WorldView(backlog, _, _, alive, pending, time) if alive.nonEmpty =>
          val stale = (alive -- pending.keys).collect({
            case (n, started) if backlog == 0 && timediff(started, time).toMinutes % 60 >= 58 => n
            case (n, started) if timediff(started, time) >= 5.hours => n
          }).toList
          NonEmptyList.fromList(stale)
        case _ =>
          None
      }
    }
  }
  object DynAgents {
    type OpTypes = _root_.iota.KList.Op.Concat[Drone.OpTypes, Machines.OpTypes]
    type Op[AA$3] = _root_.iota.CopK[OpTypes, AA$3]
    class To[GG$2[_]](implicit val d: Drone[GG$2], val m: Machines[GG$2]) extends DynAgents[GG$2] {}
    implicit def to[GG$2[_]](implicit d: Drone[GG$2], m: Machines[GG$2]): To[GG$2] = new To[GG$2]()
    def apply[GG$2[_]](implicit ev: DynAgents[GG$2]): DynAgents[GG$2] = ev
  }
}
```

**Comment by fommil** [comment]
looks like https://github.com/scalameta/paradise/issues/146

**Comment by diesalbla** [comment]
@fommil I am afraid that the [file link that you mentioned above](https://github.com/fommil/drone-dynamic-agents/blob/freestyle-bug-for/src/main/scala/logic.scala), is no longer available.
If you have another link to it, that would help us on dealing with this issue.

**Comment by fommil** [comment]
oh, I moved the project to gitlab :smile: but I think I might also have deleted the branch.

This is actually a bug in `scala.meta` that Eugene is aware of, so it should be closed here.
